src/main/java/com/elixr/springbootfile/controller/Controller.java

```diff
@@ -0,0 +1,43 @@
+package com.elixr.springbootfile.controller;
+
+
+import com.elixr.springbootfile.constants.Constants;
+import com.elixr.springbootfile.dao.ModelFile;
+import com.elixr.springbootfile.exceptionhandling.NotFoundException;
+import com.elixr.springbootfile.service.FileService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.validation.constraints.NotEmpty;
+
+
+@RestController
+@AllArgsConstructor
+@Validated
+public class Controller {
+    private final FileService fileService;
+
+    @PostMapping("/upload")
+    public ResponseEntity<?> createFile(@RequestParam("file") MultipartFile file, ModelFile modelFile, @RequestParam(value = "userName") @NotEmpty String userName) {
```

Comment:
Why do you need this ModelFile parameter in the controller?

Reply:
In the ServiceImpl class I have used the Modelfile parameter, to call that method I have used modelfile ,now I removed that and created object of modelfile in serviceImpl.